src/frontend/src/pages/FinancePage/FinanceComponents/RefundModal.tsx

```diff
@@ -0,0 +1,132 @@
+import { yupResolver } from '@hookform/resolvers/yup';
+import AttachMoneyIcon from '@mui/icons-material/AttachMoney';
+import { FormControl, FormLabel } from '@mui/material';
+import { DatePicker } from '@mui/x-date-pickers';
+import { Controller, useForm } from 'react-hook-form';
+import * as yup from 'yup';
+import LoadingIndicator from '../../../components/LoadingIndicator';
+import NERFormModal from '../../../components/NERFormModal';
+import ReactHookTextField from '../../../components/ReactHookTextField';
+import { useToast } from '../../../hooks/toasts.hooks';
+
+const schema = yup.object().shape({
+  refundAmount: yup
+    .number()
+    .required('Refund amount is required')
+    .positive('Refund amount must be positive')
+    .test('two decimals', 'Refund amount must have at most two decimal places', (value) => {
+      // technically this allows trailing zeros, but that's fine
+      if (!value) return false;
+
+      const strValue = value.toString();
+      const parts = strValue.split('.');
+      if (parts.length === 1) return true;
+      return parts[1].length <= 2;
+    })
+    .typeError('Refund amount is required'),
+  dateReceived: yup.date().required()
+});
+
+interface RefundModalProps {
+  showModal: boolean;
+  handleClose: () => void;
+  mutateAsync: (data: any) => void; // if anyone can figure out how to change this without raising type errors, be my guest
```

Comment:
what did you want to change it to? Like whats wrong with this

Reply:
Just that it uses `any`, but I think I know how to fix it now